GVFS/GVFS.Common/Prefetch/BlobPrefetcher.cs

```diff
@@ -177,7 +177,8 @@ public virtual void Prefetch(string branchOrCommit, bool isBranch)
             bool readFilesAfterDownload,
             out int matchedBlobCount,
             out int downloadedBlobCount,
-            out int readFileCount)
+            out int readFileCount,
+            bool force = false)

```

Comment:
Seeing the force flag passed around everywhere makes me think that a slight redesign is in order here.

It's actually kind of meaningless to pass in `force=true` when dealing with the `BlobPrefetcher` directly. That flag only makes sense when doing a checkout, via fastfetch, so it seems like it should only be a parameter to the constructor of `CheckoutPrefetcher`. That constructor can then pass a flag to the base constructor telling it to ignore the shallow file, but no caller of `BlobPrefetcher` should have access to that.